File: c/c++/NIM_ARRAY.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main() {
    string nim;
    cout << "Masukkan NIM: ";
    cin >> nim;

    // ubah string menjadi array digit
    vector<int> digit;
    for (int i = 0; i < nim.length(); i++) {
        digit.push_back(nim[i] - '0');
    }

    // Total
    int total = 0;
    for (int d : digit) {
        total += d;
    }

    // Maksimum
    int maks = digit[0];
    for (int d : digit) {
        if (d > maks) {
            maks = d;
        }
    }

    // Minimum
    int minin = digit[0];
    for (int d : digit) {
        if (d < minin) {
            minin = d;
        }
    }

    // Rata-rata
    float rata = (float) total / digit.size();

    // Reverse array
    vector<int> rev;
    for (int i = digit.size() - 1; i >= 0; i--) {
        rev.push_back(digit[i]);
    }

    // Output
    cout << "Digit : ";
    for (int d : digit) {
        cout << d << " ";
    }
    cout << endl;

    cout << "Total = " << total << endl;
    cout << "Maksimum = " << maks << endl;
    cout << "Minimum = " << minin << endl;
    cout << "Rata-rata = " << rata << endl;

    cout << "Reverse array: ";
    for (int d : rev) {
        cout << d << " ";
    }
    cout << endl;

    return 0;
}

File: php/NIM_ARRAY.php
<?php
echo "Masukkan NIM: ";
$nim = trim(fgets(STDIN));

# ubah string menjadi array digit
$digit = [];
for ($i = 0; $i < strlen($nim); $i++) {
    $digit[] = (int)$nim[$i];
}

# Total
$total = 0;
foreach ($digit as $d) {
    $total += $d;
}

# Maksimum
$maks = $digit[0];
foreach ($digit as $d) {
    if ($d > $maks) {
        $maks = $d;
    }
}

# Minimum
$minin = $digit[0];
foreach ($digit as $d) {
    if ($d < $minin) {
        $minin = $d;
    }
}

# Rata-rata
$rata = $total / count($digit);

# Reverse array
$rev = [];
for ($i = count($digit) - 1; $i >= 0; $i--) {
    $rev[] = $digit[$i];
}

# Output
echo "Digit : ";
foreach ($digit as $d) {
    echo $d . " ";
}
echo PHP_EOL;

echo "Total = $total" . PHP_EOL;
echo "Maksimum = $maks" . PHP_EOL;
echo "Minimum = $minin" . PHP_EOL;
echo "Rata-rata = $rata" . PHP_EOL;

echo "Reverse array: ";
foreach ($rev as $d) {
    echo $d . " ";
}
echo PHP_EOL;
